### PR TITLE
Added swipe to cancel functionality on the batch jobs #137

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

https://github.com/hotwax/job-manager/issues/137

Closes #
https://github.com/hotwax/job-manager/commit/cfcf91e29b8670f98ae9a1211256c24ce583e715

### Short Description and Why It's Useful

Now user will easily be able to delete a batch by just swiping and confirming to delete it.

### Functionality screenshots
Screenshot 2022-05-16 at 3 24 45 PM
Screenshot 2022-05-16 at 3 24 32 PM

### Video demonstration

https://user-images.githubusercontent.com/62360781/168566868-266db50a-46e3-48a0-b41e-514fea61b303.mov

### CODE REVIEW

**Comment by ymaheshwari1** [comment]
@shashwatbanPlease add a screenshot and a video of the functionality.

File: src/views/Orders.vue [hunk 1]
```diff
@@ -212,6 +226,25 @@ export default defineComponent({
       });
       return batchmodal.present();
     },
+    async deleteBatch(batch: any) {
```
Comment (dt2patel) [inline src/views/Orders.vue:229]: We should have global function that cancels jobs right? Why do we have to define new method here?

Can we not link method directly to action?

File: src/views/Orders.vue [hunk 2]
```diff
@@ -212,6 +226,25 @@ export default defineComponent({
...
+      const deleteBatchAlert = await alertController
+        .create({
+          message: this.$t('Are you sure?'),
```
Comment (dt2patel) [inline src/views/Orders.vue:232]: Did not use correct alert here like we discussed.

**Review by dt2patel — CHANGES_REQUESTED** [review]
Please see comments

**Comment by shashwatbangar** [comment]
> @shashwatbanPlease add a screenshot and a video of the functionality.

Done sir!

File: src/views/Orders.vue (hunk 1 above)
Comment (adityasharma7) [reply src/views/Orders.vue:229]: As we are calling action for delete button of alert.
https://github.com/hotwax/job-manager/pull/140/files#diff-2dfe4d12b346e93d35ad5538343555117ecc5df00c7ad7c067d945535c696d44R241

We could mark it resolved

File: src/views/Orders.vue (hunk 2 above)
Comment (adityasharma7) [reply src/views/Orders.vue:232]: Reference:
https://github.com/hotwax/job-manager/blob/main/src/views/Pipeline.vue#L446

Comment (shashwatbangar) [reply src/views/Orders.vue:232]: > Reference:
> https://github.com/hotwax/job-manager/blob/main/src/views/Pipeline.vue#L446

Thank you sir! I am referring this link.

File: src/views/Orders.vue [hunk 3]
```diff
@@ -100,10 +100,18 @@
...
-            </ion-item>
+
+            <ion-item-sliding  v-for="batch in getJob(jobEnums['BTCH_BRKR_ORD'])" :key="batch?.id" button detail v-show="batch?.status === 'SERVICE_PENDING'">
+              <ion-item  @click="editBatch(batch?.id)">
+                <ion-label class="ion-text-wrap">{{ batch?.jobName }}</ion-label>
+                <ion-note slot="end">{{ batch?.runTime ? getTime(batch.runTime) : '' }}</ion-note>
+              </ion-item>
+              <ion-item-options side="end">
+                <ion-item-option @click="deleteBatch(batch)" color="danger">
+                  <ion-icon slot="icon-only" :icon="trash"></ion-icon>
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:111]: Use self-closing tag.

File: src/views/Orders.vue [hunk 4]
```diff
@@ -100,10 +100,18 @@
                 <ion-icon :icon="addCircleOutline" slot="end" />
               </ion-button>
             </ion-item-divider>
-            <ion-item v-for="batch in getJob(jobEnums['BTCH_BRKR_ORD'])" :key="batch?.id" button detail @click="editBatch(batch?.id)" v-show="batch?.status === 'SERVICE_PENDING'">
-              <ion-label class="ion-text-wrap">{{ batch?.jobName }}</ion-label>
-              <ion-note slot="end">{{ batch?.runTime ? getTime(batch.runTime) : '' }}</ion-note>
-            </ion-item>
+
+            <ion-item-sliding  v-for="batch in getJob(jobEnums['BTCH_BRKR_ORD'])" :key="batch?.id" button detail v-show="batch?.status === 'SERVICE_PENDING'">
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:104]:
```suggestion
            <ion-item-sliding v-for="batch in getJob(jobEnums['BTCH_BRKR_ORD'])" :key="batch?.id" button detail v-show="batch?.status === 'SERVICE_PENDING'">
```

File: src/views/Orders.vue [hunk 5]
```diff
@@ -100,10 +100,18 @@
...
+              <ion-item  @click="editBatch(batch?.id)">
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:105]:
```suggestion
              <ion-item @click="editBatch(batch?.id)">
```

File: src/views/Orders.vue [hunk 6]
```diff
@@ -282,7 +319,7 @@ export default defineComponent({
         });
 
       return jobAlert.present();
-    }
+    },
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:322]: Remove unwanted change

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Orders.vue (hunk 6 above)
Comment (shashwatbangar) [reply src/views/Orders.vue:322]: Okay sir!

Comment (shashwatbangar) [reply src/views/Orders.vue:319]: @ymaheshwari1 sir, I've made all the changes as said, Please review
https://github.com/hotwax/job-manager/pull/140/commits/5b4e8884f2999fb2614dcd5679e0cbb6622cdfee

File: src/views/Orders.vue [hunk 7]
```diff
@@ -212,6 +226,26 @@ export default defineComponent({
...
+          message: this.$t("Canceling this job will cancel this occurrence and all following occurrences. This job will have to be re-enabled manually to run it again."),
+          buttons: [
+            {
+              text: this.$t("Don't cancel"),
+              role: 'cancel',
+            },
+            {
+              text: this.$t("Cancel"),
+              handler: async () => {
+                  await this.store.dispatch('job/cancelJob', batch);
```
Comment (ymaheshwari1) [inline src/views/Orders.vue:242]: Improve indentation to one tab/two space.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/Orders.vue (hunk 7 above)
Comment (shashwatbangar) [reply src/views/Orders.vue]: Done sir!
Please review - https://github.com/hotwax/job-manager/pull/140/commits/989d77e24f83a00ba4ec6ff6ae5ad50dcef77bf0

File: src/views/Orders.vue [hunk 8]
```diff
@@ -280,9 +314,8 @@ export default defineComponent({
             }
           ]
         });
-
       return jobAlert.present();
-    }
+    },
```
Comment (adityasharma7) [inline src/views/Orders.vue:318]:
```suggestion
    }
```

File: src/locales/en.json [hunk 9]
```diff
@@ -38,11 +38,13 @@
   "Daily": "Daily",
   "Dashoard": "Dashoard",
   "Days": "Days",
+  "Delete": "Delete",
   "Disable": "Disable",
   "Dismiss": "Dismiss",
   "Disabled": "Disabled",
   "Discard changes": "Discard changes",
   "Don't cancel": "Don't cancel",
+  "Don't delete": "Don't delete",
```
Comment (adityasharma7) [inline src/locales/en.json:47]: I think we could revert these labels as they are not used anymore

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**